features/cellular/TARGET_GENERIC_MODEM/generic_modem_driver/ReferenceCellularInterface.h

```diff
@@ -0,0 +1,271 @@
+    CSD_UNKNOWN_COVERAGE=4,
+    CSD_REGISTERED_ROAMING=5,
+    CSD_SMS_ONLY=6,
+    CSD_SMS_ONLY_ROAMING=7,
+    CSD_CSFB_NOT_PREFERRED=9
+} nwk_registration_status_csd;
+
+/**
+ * Packet Switched network registration status (CGREG Usage)
+ * UBX-13001820 - AT Commands Example Application Note (Section 18.27.3)
+ */
+typedef enum {
+    PSD_NOT_REGISTERED_NOT_SEARCHING=0,
+    PSD_REGISTERED=1,
+    PSD_NOT_REGISTERED_SEARCHING=2,
+    PSD_REGISTRATION_DENIED=3,
+    PSD_UNKNOWN_COVERAGE=4,
+    PSD_REGISTERED_ROAMING=5,
+    PSD_EMERGENCY_SERVICES_ONLY=8
+} nwk_registration_status_psd;
+
+typedef struct {
+    char ccid[20+1];    //!< Integrated Circuit Card ID
+    char imsi[15+1];    //!< International Mobile Station Identity
+    char imei[15+1];    //!< International Mobile Equipment Identity
+    char meid[18+1];    //!< Mobile Equipment IDentifier
+    int flags;
+    bool ppp_connection_up;
+    radio_access_nwk_type rat;
+    nwk_registration_status_csd reg_status_csd;
+    nwk_registration_status_psd reg_status_psd;
+} device_info;
+
+/** ReferenceCellularInterface class
+ *
+ *  This interface serves as the controller/driver for the Cellular
+ *  modems (tested with UBLOX_C027 and MTS_DRAGONFLY_F411RE).
+ */
+class ReferenceCellularInterface : public CellularInterface {
+
+public:
+    ReferenceCellularInterface(bool debugOn = false, PinName tx = MDMTXD, PinName rx = MDMRXD, int baud = MBED_CONF_REF_CELL_DRV_BAUD_RATE);
+    ~ReferenceCellularInterface();
+
+    /** Set the Cellular network credentials
+     *
+     *  Please check documentation of connect() for default behaviour of APN settings.
+     *
+     *  @param apn      Access point name
+     *  @param uname    optionally, Username
+     *  @param pwd      optionally, password
+     */
+    virtual void set_credentials(const char *apn, const char *uname = 0,
+                                                  const char *pwd = 0);
+
+    /** Set the pin code for SIM card
+     *
+     *  @param sim_pin      PIN for the SIM card
+     */
+    virtual void set_sim_pin(const char *sim_pin);
+
+    /** Start the interface
+     *
+     *  Attempts to connect to a Cellular network.
+     *  This driver is written mainly for data network connections as CellularInterface
+     *  is NetworkInterface. That's whu connect() call internaly calls nwk_registration()
+     *  method with parameter PACKET_SWITCHED network. Circuit switched hook and registration
+     *  process is implemented and left in the driver for future extension/subclass support,e.g.,
+     *  an SMS or GPS extension.
+     *
+     *  @param sim_pin     PIN for the SIM card
+     *  @param apn         optionally, access point name
+     *  @param uname       optionally, Username
+     *  @param pwd         optionally, password
+     *  @return            NSAPI_ERROR_OK on success, or negative error code on failure
+     */
+    virtual nsapi_error_t connect(const char *sim_pin, const char *apn = 0,
```

Comment:
Ah found the changes here:
https://github.com/ARMmbed/mbed-os/pull/4119/files/bd37b7c00fe6de549d7424436bcfe979cabcd0af#diff-a1349b6ff207a9467417014cc6d2f4ccL54

I am for this change, although it may break existing cellular drivers. If we are being strict on backwards compatibility we should keep the old api around with a deprecation notice.